### PR TITLE
Make cmake binary configurable

### PR DESCRIPTION
**Pull request by jcftang**
I needed this on centos7 with epel, the cmake version 3 binary is named cmake3 and not cmake.

### CODE REVIEW

File: bin/polly.py [hunk 1]
```diff
@@ -196,6 +196,11 @@ def PositiveInt(string):
     help='Timeout for CTest'
 )
 
+parser.add_argument(
+    '--cmake',
+    help="CMake binary (cmake or cmake3)",
```
Comment (ruslo) [inline bin/polly.py]: `,` at the end is redundant

File: bin/polly.py [hunk 2]
```diff
@@ -309,19 +314,24 @@ def PositiveInt(string):
     cdir, args.verbosity, args.discard, args.tail, polly_toolchain
 )
 
+if args.cmake:
+    CMAKE_BIN = args.cmake
```
Comment (ruslo) [inline bin/polly.py]: 2 spaces indent

File: bin/polly.py [hunk 3]
```diff
@@ -309,19 +314,24 @@ def PositiveInt(string):
...
+else:
+    CMAKE_BIN = 'cmake'
```
Comment (ruslo) [inline bin/polly.py]: please rename to `cmake_bin`, `CMAKE_BIN` assume it something global

File: bin/polly.py [hunk 4]
```diff
@@ -309,19 +314,24 @@ def PositiveInt(string):
...
+
 if os.name == 'nt':
   # Windows
-  detail.call.call(['where', 'cmake'], logging)
+  detail.call.call(['where', CMAKE_BIN], logging)
```
Comment (ruslo) [inline bin/polly.py]: If `CMAKE_BIN` is a full path to `cmake` just check that this file exists.

Comment (jcftang) [inline bin/polly.py]: I wasn't too sure what you meant by this, so I added the check just before this if block

File: bin/polly.py [hunk 5]
```diff
@@ -309,19 +314,28 @@ def PositiveInt(string):
...
+if args.cmake:
+  cmake_bin = args.cmake
+else:
+  cmake_bin = 'cmake'
+
+if os.path.isabs(cmake_bin):
+  if not os.path.exists(cmake_bin):
+    sys.exit("CMake binary not found: {}".format(cmake_bin))
+
 if os.name == 'nt':
```
Comment (ruslo) [inline bin/polly.py]: if `cmake_bin` is absolute path then it make no sense to run `where`/`which`. Actually `where` will not work. Should be like:
```python
if os.path.isabs(cmake_bin):
  # check exists
else:
  if os.name == 'nt':
    # where
  else:
    # which
```

**Comment by jcftang** [comment]
I did a bit more testing in a clean environment and fixed a case where I missed a call to cmake

**Pull request merged**